src/poetry/puzzle/provider.py

```diff
@@ -560,11 +559,6 @@ def complete_package(
                     if locked is not None and locked.package.is_same_package_as(dep):
                         continue
                     self.search_for_direct_origin_dependency(dep)
-        else:
-            for dep in _dependencies:
-                if dep.is_file() or dep.is_directory():
-                    dep = cast("PathDependency", dep)
-                    dep.validate(raise_error=True)
```

Comment:
Yes:

```
❯ cd tests/fixtures/directory/project_with_transitive_directory_dependencies_missing 
❯ poetry lock                      
Path poetry/tests/fixtures/directory/foo for foo does not exist
Updating dependencies
Resolving dependencies... (0.0s)

Path poetry/tests/fixtures/directory/foo for foo does not exist
```

But should we add a test for it? It's not clear to me if that's covered by `installer.run` or not (since it seems that `poetry lock` uses `installer.lock()` + `installer.run()` under the hood or something.

Reply:
It's probably covered, but to be sure we could add a test to test_lock similar to the tests in #6844.

Reply:
I believe we added this exact test in https://github.com/python-poetry/poetry/blob/e334f524152588886ac6e4e075190bdb16ad344d/tests/console/commands/test_lock.py#L207-L232

Reply:
I was just wondering if "project_with_**transitive**_directory_dependencies_missing" is something special. Do we even need the new fixture or is it just the same as missing directory dependency?

Reply:
Ok looked at both in detail and there was no real difference aside from the fact that in `missing_directory_dependency` the dependency was a dev dependency (and was being used to check that if you skip a group that has a missing dependency installation works). So we still need two fixtures, but I renamed them a bit to make it clearer what the difference is.

Reply:
I don't think we need two fixtures because it doesn't matter if the missing dependency is in a group or not if the group is not optional and we don't pass any flags to the installer to skip the group.